core/src/main/java/org/mule/routing/AbstractUntilSuccessfulProcessingStrategy.java

```diff
@@ -96,4 +98,45 @@ protected UntilSuccessfulConfiguration getUntilSuccessfulConfiguration()
         return untilSuccessfulConfiguration;
     }
 
+    @Override
+    public MuleEvent route(MuleEvent event) throws MessagingException
+    {
+        prepareAndValidateEvent(event);
+        return doRoute(event);
+    }
+
+    protected abstract MuleEvent doRoute(final MuleEvent event) throws MessagingException;
+
+    private void prepareAndValidateEvent(final MuleEvent event) throws MessagingException
+    {
+        try
+        {
+            final MuleMessage message = event.getMessage();
+            if (message instanceof DefaultMuleMessage)
+            {
+                if (((DefaultMuleMessage) message).isConsumable())
+                {
+                    message.getPayloadAsBytes();
+                }
+                else
+                {
+                    validateNonConsumableMessage(message);
+                }
+            }
+            else
+            {
+                message.getPayloadAsBytes();
+            }
+        }
+        catch (final Exception e)
+        {
+            throw new MessagingException(MessageFactory.createStaticMessage("Failed to prepare message for processing"), event, e, getUntilSuccessfulConfiguration().getRouter());
+        }
+    }
+
+    protected void validateNonConsumableMessage(final MuleMessage message) throws Exception
```

Comment:
This method's name is unclear... you can potentially need to do more than just validating here... Additionally, this should receive the event, not just the message